Source/WebCore/bindings/js/JSImageDataCustom.cpp

```diff
@@ -43,10 +43,12 @@ JSValue toJSNewlyCreated(JSGlobalObject* lexicalGlobalObject, JSDOMGlobalObject*
     auto& data = imageData->data();
     auto* wrapper = createWrapper<ImageData>(globalObject, WTFMove(imageData));
     Identifier dataName = Identifier::fromString(vm, "data"_s);
-    wrapper->putDirect(vm, dataName, toJS(lexicalGlobalObject, globalObject, data), PropertyAttribute::DontDelete | PropertyAttribute::ReadOnly);
+    wrapper->putDirect(vm, dataName, data.visit([&](auto& array) {
+        return toJS(lexicalGlobalObject, globalObject, array);
+    }), PropertyAttribute::DontDelete | PropertyAttribute::ReadOnly);
```

Comment:
I do not think this change is needed. `toJS()` above calls this function:
```
inline JSC::JSValue toJS(JSC::JSGlobalObject* lexicalGlobalObject, JSC::JSGlobalObject* globalObject, JSC::ArrayBufferView& view)
{
    return view.wrap(lexicalGlobalObject, globalObject);
}
```
So whether you pass `Uint8ClampedArray` or `Float16Array`, it will be casted to the base class `ArrayBufferView`. So I think this can be changed to 
```
auto& data = imageData->data().getArrayBufferView();
wrapper->putDirect(vm, dataName, toJS(lexicalGlobalObject, globalObject, data), PropertyAttribute::DontDelete | PropertyAttribute::ReadOnly);
```